plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php

```diff
@@ -612,7 +612,7 @@ private function get_hpos_setting_for_feature() {
 
 		$get_disabled = function () {
 			$compatibility_info = $this->features_controller->get_compatible_plugins_for_feature( 'custom_order_tables', true );
-			$sync_complete      = 0 === $this->data_synchronizer->get_current_orders_pending_sync_count();
+			$sync_complete      = ! $this->data_synchronizer->has_orders_pending_sync();
```

Comment:
A number of these places are not using the first argument of `$use_cache` and result in re-querying the pending orders each time.

I suppose this is probably necessary to keep up-to-date information for switching between storage versions.

This is less of a problem with the new query that is significantly more performant, but if we wanted to go a step further, we could rely on cached values and bust this case on order create/update/delete.  However, I was a little hesitant to do this since this count is important to avoid switching storage options which could result in data corruption if orders are pending.